File: python/1.py
data = open('input.txt', 'r').read().split('\n')

for number1 in data:
  for number2 in data:
    if ((int(number1) + int(number2)) == 2020):
      print("%s * %s = %i" % (number1, number2, int(number1)*int(number2)))

for number1 in data:
  for number2 in data:
    for number3 in data:
      if ((int(number1) + int(number2) + int(number3)) == 2020):
        print("%s * %s * %s = %i" % (number1, number2, number3, int(number1)*int(number2)*int(number3)))

File: 2/1.js
import fs from 'fs';

function isValid({min, max, letter, password}) {
  let count = 0; 
  for (const c of password) {
    if (c == letter) count++; 
  }
  if (min <= count && count <= max) return true; 
  return false; 
}

export default function solve() {

  let data = fs.readFileSync('2/input.txt', 'utf8')
    .split("\r\n")
    .map(line => {
      let arr = line.split(/-|: | /g);
      return {
        min: arr[0],
        max: arr[1],
        letter: arr[2],
        password: arr[3]
      }
    });

  let count = 0; 
  data.forEach(e => {
    if (isValid(e)) count++; 
  });
  
  console.log(count);
}